tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNet/ControllerActionInvoker_InvokeAction_Integration.cs

```diff
@@ -0,0 +1,80 @@
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AspNet
+{
+    /// <summary>
+    /// System.Web.Mvc.Async.AsyncControllerActionInvoker.BeginInvokeAction calltarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = AssemblyName,
+        TypeName = "System.Web.Mvc.ControllerActionInvoker",
+        MethodName = "InvokeActionMethod",
+        ReturnTypeName = ActionResultTypeName,
+        ParameterTypeNames = new[] { ControllerContextTypeName, ActionDescriptorTypeName, DictionaryTypeName },
+        MinimumVersion = MinimumVersion,
+        MaximumVersion = MaximumVersion,
+        IntegrationName = IntegrationName)]
+    // ReSharper disable once InconsistentNaming
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class ControllerActionInvoker_InvokeAction_Integration
+    {
+        private const string AssemblyName = "System.Web.Mvc";
+        private const string ActionResultTypeName = "System.Web.Mvc.ActionResult";
+        private const string ControllerContextTypeName = "System.Web.Mvc.ControllerContext";
+        private const string ActionDescriptorTypeName = "System.Web.Mvc.ActionDescriptor";
+        private const string DictionaryTypeName = "System.Collections.Generic.IDictionary`2[System.String,System.Object]";
+        private const string MinimumVersion = "4";
+        private const string MaximumVersion = "5";
+
+        private const string IntegrationName = nameof(IntegrationId.AspNetMvc);
+
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<ControllerActionInvoker_InvokeAction_Integration>();
+
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TContext">Controller context</typeparam>
+        /// <typeparam name="TActionDescriptor">Action descriptor</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="controllerContext">The control context instance</param>
+        /// <param name="actionDescriptor">The action descriptor instance</param>
+        /// <param name="parameters">The parameters of the mvc method</param>
+        /// <returns>Calltarget state value</returns>
+        internal static CallTargetState OnMethodBegin<TTarget, TContext, TActionDescriptor>(TTarget instance, TContext controllerContext, TActionDescriptor actionDescriptor, IDictionary<string, object> parameters)
+        {
+            try
+            {
+                var security = Security.Instance;
+                var context = HttpContext.Current;
+                if (context != null && security.Settings.Enabled)
+                {
+                    var scope = SharedItems.TryPeakScope(context, AspNetMvcIntegration.HttpContextKey);
```

Comment:
```suggestion
                    var scope = SharedItems.TryPeekScope(context, AspNetMvcIntegration.HttpContextKey);
```

Reply:
Fixed in https://github.com/DataDog/dd-trace-dotnet/pull/2495/commits/6785f56e8af6d600e01818116fa36b76c7e0cdb6